futureSquash/src/test/scala/FutureSquashSpec.scala

```diff
@@ -0,0 +1,124 @@
+import org.scalatest._
+import scala.concurrent.{ExecutionContext, _}
+import scala.util._
+import com.softwaremill.futuresquash._
+
+class FutureSquashSpec extends AsyncFlatSpec with Matchers {
+
+  override implicit val executionContext = ExecutionContext.global
+
+  import FutureSquash._
+
+  def fea: Future[Either[Throwable, Int]] = Future(Right(1))
+
+  def feb(a: Int): Future[Either[Throwable, String]] = Future(Right(s"${a}b"))
+
+  abstract class Error(message: String) extends Exception(message)
+
+  case object BoomError extends Error("Boom")
+
+  "FutureSquash" should "convert Future[Right[Throwable, A]] to Future[A]" in {
+    FutureSquash.fromEither(Right("a")) map {
+      _ shouldBe "a"
+    }   
+  }
+
+  "FutureSquash" should "convert Future[Left[Throwable, A]] to Future[A]" in {
+    FutureSquash.fromEither(Left(BoomError)).failed map {
+      _ shouldBe BoomError
+    }
+  }
+
+  "FutureSquash" should "squash Future[Right[Throwable, A]]" in {
```

Comment:
Test cases that use for-comprehension with two `squash`ed `Future`s test `Future`. I think test cases with only `FutureSqueeze` is enough, but not a big problem to leave it here.